### PR TITLE
Internal Dashboards

### PR DESCRIPTION
**Pull request by jessleenyc**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description

For the first time ever, my commits are pretty reflective of all the changes in this PR. Basically, the goal was to quickly remove fields from /admin but when I did, codeclimate was screaming at me about a few things so I ended up moving a bunch of code around.

- Creates route/module? for updating user roles (ban, warned, pro, mentorship, etc) 
- Removes a bunch of fields from admin such as tag moderator id, scholar, banned, warned, etc.
(and either updated or removed the associated spec)
- /moderate now redirect to internal user show page instead of /edit 
- Fixes user profile not updating when banishing a user for spam
- Added a note creation service
- Moves activity and note into a partial view since they're helpful info on both show and edit pages for internal user..

..and more, look at the commits!

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [ ] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/services/note_creation_service.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class NoteCreationService
```
Comment (maestromac) [inline app/services/note_creation_service.rb]: This class is kind of unnecessary right now. A simple `Note.create(..)` in the needed place is enough

File: app/services/moderator/banish_user.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
+module Moderator
+  class BanishUser < ManageActivityAndRoles
```
Comment (maestromac) [inline app/services/moderator/banish_user.rb]: Should make some of these method as private method.

File: app/views/internal/users/show.html.erb [hunk 3]
```diff
@@ -36,65 +36,8 @@
...
-    <%= f.label "Warn User" %>
-    <%= f.check_box :warn_user, checked: @user.warned %>
-    <br>
-    <%= f.label "Reason for action:" %>
-    <%= f.text_area :note_for_current_role, required: true %>
-    <br>
-    <%= f.submit "Update User Status" %>
-  <% end %>
 </div>
+<%= render "internal/users/activity" %>
```
Comment (maestromac) [inline app/views/internal/users/show.html.erb]: Since `activity` is a partial, this should work right?
```suggestion
<%= render "activity" %>
```

Comment (benhalpern) [reply app/views/internal/users/show.html.erb]: Yeah, as long as the scoping is appropriate. Sometimes we use partials from different scopes, but this should be fine.

**Pull request merged**